python/pyspark/install.py

```diff
@@ -65,7 +68,7 @@ def checked_versions(spark_version, hadoop_version, hive_version):
 
     if hadoop_version == "without":
         hadoop_version = "without-hadoop"
-    elif re.match("^[0-9]+\\.[0-9]+$", hadoop_version):
+    elif re.match("^[0-9]+$", hadoop_version):
```

Comment:
I'm wondering whether we should support installing older versions of Spark here (cc @HyukjinKwon). If so, we might need to also include `hadoop-2.7` etc as valid inputs?

Reply:
Oh, I think actually @AngersZhuuuu covered old version logics too.  There are some tests at `test_install_spark.py` below. For testing, it always use old versions. For releases, it always uses the corresponding version string (e.g., if pip install pyspark==3.2.0, `spark_version` is always `3.2.0`)

Reply:
Ah saw that. So for `spark_version = 3.2.0` it'll translate `hadoop_version` = `hadoop3` into `hadoop3.2`.